debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -110,7 +132,24 @@ public static ConnectionFactory patternBasedFactory(String urlPattern, Field...
             String url = findAndReplace(urlPattern, props, varsWithDefaults);
             LOGGER.trace("Props: {}", props);
             LOGGER.trace("URL: {}", url);
-            Connection conn = DriverManager.getConnection(url, props);
+            
+            Connection conn = null;
+            ClassLoader driverClassLoader = classloader;
+            String driverClassName = config.getString(JdbcConfiguration.DRIVER);
+            if (driverClassName == null) {
+                conn = DriverManager.getConnection(url, props);
+            } else {
+                try {
+                    if (driverClassLoader == null) {
+                        driverClassLoader = JdbcConnection.class.getClassLoader();
+                    }
+                    Class<java.sql.Driver> driverClazz = (Class<java.sql.Driver>) Class.forName(driverClassName, true, driverClassLoader);
+                    java.sql.Driver driver = driverClazz.newInstance();
+                    conn = driver.connect(url, props);
+                } catch (ClassNotFoundException|IllegalAccessException|InstantiationException e) {
+                    throw new SQLException(e);
+                }                
+            }
```

Comment:
This whole method is relatively complex now. It really would be simpler if the old `patternBasedFactory(String urlPattern, Field...fields)` method did what it used to do (using `DriverManager.getConnection(url, props)`) and this method always used the `driverClassName` provided as a parameter and created an instance. Yes, it duplicates a bit of code, but each method is much more direct/obvious and there are fewer possible paths through each method.